authorization/role/model/role_management_model_service.go

```diff
@@ -38,6 +40,32 @@ type GormRoleManagementModelService struct {
 	repository repository.Repositories
 }
 
+// ListAssignmentsByIdentityAndResource lists all the roles that have been assigned to a specific user/identity for a specific resource
+func (r *GormRoleManagementModelService) ListAssignmentsByIdentityAndResource(ctx context.Context, resourceID string, identityID uuid.UUID) ([]identityrole.IdentityRole, error) {
+	return r.repository.IdentityRoleRepository().ListByIdentityAndResource(ctx, resourceID, identityID)
+}
+
+// Assign assigns a user identity with a role, for a specific resource
+func (r *GormRoleManagementModelService) Assign(ctx context.Context, identityID uuid.UUID, resourceID string, roleName string) error {
+
+	rt, err := r.repository.ResourceRepository().Load(ctx, resourceID)
+	if err != nil {
+		return err
+	}
+
+	roleRef, err := r.repository.RoleRepository().Lookup(ctx, roleName, rt.ResourceType.Name)
```

Comment:
I believe this is because the role name is being passed in, and a lookup must be done on the role table to find the actual role ID for that resource type (which needs to be looked up also).  We could conceivably reduce the number of DB calls by executing a raw query which joins the `resource` and `role` tables by `resource_type_id` and obtaining the `role_id` that way.  This could go inside a helper method somewhere, possibly in the role repository type.

Reply:
I just noticed there's also still a .Debug() statement at line 185 in this file.

Reply:
Correct, I was thinking of something radical. Should we consider getting rid of the role_id and use the ( name , resource_type ) as the primary key ? Because almost everywhere we are going to encourage use of the role name, instead of the role ID.